dataService.js

```diff
@@ -0,0 +1,71 @@
+// Required Maps
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+/*
+ * Method to create and return map of products if its not already created
+ */
+function getProducts() {
+
+  if (!productMap || !productMap.size) {
+    const productData = require('./src/data/products.json');
+    productMap = {};
+
+    productData.products.forEach(product => {
+      productMap[product.id] = product;
```

Comment:
Rather than modifying the global variable inside the loop, Can you create a variable inside the method and create the map.
At the end of the loop, Can you assign that to `productMap`?

Reply:
Done :)